### PR TITLE
[Y2K-570] SignupScreen'd relogin

### PR DESCRIPTION
**Pull request by jzila**
after:
Screen Shot 2019-09-09 at 11 27 16 AM

before:
Screen Shot 2019-09-09 at 11 25 45 AM

- [ ] stories before merge

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/login/relogin/index.desktop.tsx [hunk 1]
```diff
@@ -51,89 +52,150 @@ class Login extends React.Component<Props, State> {
...
             />
-            <Kb.FormWithCheckbox
-              style={{alignSelf: 'stretch'}}
-              inputProps={inputProps}
-              checkboxesProps={checkboxProps}
-            />
-            <Kb.WaitingButton
-              disabled={!this.props.password}
+            <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.inputRow}>
+              <Kb.Input {...inputProps} />
```
Comment (pzduniak) [inline shared/login/relogin/index.desktop.tsx]: new mocks use Kb.LabeledInput

Comment (jzila) [reply shared/login/relogin/index.desktop.tsx]: That excludes `passwordVisible`, is that ok?

Comment (jzila) [reply shared/login/relogin/index.desktop.tsx]: Also `LabeledInput` has buggy behavior. The user dropdown doesn't properly hover/click on a name that overlays above the `LabeledInput`, even when I change the `inputRef` to refer to a `LabeledInput`.

File: shared/common-adapters/labeled-input.tsx [hunk 2]
```diff
@@ -149,7 +149,7 @@ const styles = Styles.styleSheetCreate(
         paddingLeft: Styles.globalMargins.xsmall,
         paddingRight: Styles.globalMargins.xsmall,
         width: '100%',
-        zIndex: 10,
+        zIndex: 0,
```
Comment (jzila) [inline shared/common-adapters/labeled-input.tsx]: Not sure what's going on here, but this doesn't seem to be necessary, and it was messing up overlays.

**Comment by jzila** [comment]
OK @pzduniak that needed a lot more surgery than I liked, but at least now the wrapper is hookified.

File: shared/login/relogin/container.tsx [hunk 3]
```diff
@@ -51,9 +47,9 @@ class LoginWrapper extends React.Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props) {
-    // Clear the password when there's an input error.
-    if (this.props.inputError !== prevProps.inputError) {
-      this.setState(p => ({inputKey: p.inputKey + 1, password: ''}))
+    // Clear the password when there's an error.
+    if (this.props.error !== prevProps.error) {
+      this.setState(_ => ({password: ''}))
```
Comment (chrisnojima) [inline shared/login/relogin/container.tsx]: callback form is only used if you need the old value, just `this.setState({password: ''})`

File: shared/login/join-or-login/index.tsx [hunk 4]
```diff
@@ -15,47 +15,49 @@ type Props = Kb.PropsWithTimer<{
...
-      <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.header}>
-        <InfoIcon />
-      </Kb.Box2>
-
-      {!!props.bannerMessage && (
-        <Kb.Box2 direction="vertical" fullWidth={true} style={styles.banner}>
-          <Kb.Text center={true} type="BodySmallSemibold" style={styles.bannerMessage}>
-            {props.bannerMessage}
-          </Kb.Text>
+    <SignupScreen noBackground={true} title="Log in foo">
```
Comment (chrisnojima) [inline shared/login/join-or-login/index.tsx]: ????

File: shared/login/join-or-login/index.tsx [hunk 5]
```diff
@@ -15,47 +15,49 @@ type Props = Kb.PropsWithTimer<{
...
-      >
-        <Kb.Box2 direction="vertical" fullWidth={true} gap="small" alignItems="center">
-          <Kb.Icon type="icon-keybase-logo-64" />
-          <Kb.Text type="HeaderBig" style={styles.text}>
-            Join Keybase
-          </Kb.Text>
+        {(true || !!props.bannerMessage) && (
+          <Kb.Box2 direction="vertical" fullWidth={true} style={styles.banner}>
+            <Kb.Text center={true} type="BodySmallSemibold" style={styles.bannerMessage}>
+              {'test banner' || props.bannerMessage}
```
Comment (chrisnojima) [inline shared/login/join-or-login/index.tsx]: ?????

File: shared/login/join-or-login/index.tsx [hunk 6]
```diff
@@ -15,47 +15,49 @@ type Props = Kb.PropsWithTimer<{
...
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="small" alignItems="center">
+            <Kb.Icon type="icon-keybase-logo-64" />
+            <Kb.Text type="HeaderBig" style={styles.text}>
+              Join Keybase
+            </Kb.Text>
+          </Kb.Box2>
+          <Kb.ButtonBar direction="column" fullWidth={Styles.isMobile} style={styles.buttonBar}>
+            <Kb.Button label="Create an account" onClick={props.onSignup} fullWidth={true} />
+            <Kb.Button label="Log in" mode="Secondary" onClick={props.onLogin} fullWidth={true} />
+            {props.isOnline === false && (
```
Comment (chrisnojima) [inline shared/login/join-or-login/index.tsx]: just `!props.isOnline &&`

Comment (chrisnojima) [reply shared/login/join-or-login/index.tsx]: (i know this is from before)

**Comment by jzila** [comment]
Ready for another look @chrisnojima

File: shared/login/join-or-login/index.tsx (hunk 4 above)
Comment (chrisnojima) [reply shared/login/join-or-login/index.tsx]: why does this have 'log in foo' in the title?

File: shared/login/relogin/container.tsx [hunk 7]
```diff
@@ -32,57 +23,65 @@ type Props = {
...
 
-  componentDidUpdate(prevProps: Props) {
-    // Clear the password when there's an input error.
-    if (this.props.inputError !== prevProps.inputError) {
-      this.setState(p => ({inputKey: p.inputKey + 1, password: ''}))
+  const dispatch = Container.useDispatch()
+
+  const onSubmit = React.useCallback(() => {
+    props.onLogin(selectedUser, password)
+  }, [selectedUser, password, props])
```
Comment (chrisnojima) [inline shared/login/relogin/container.tsx]: don't have all of props be a dependent

File: shared/login/relogin/container.tsx [hunk 8]
```diff
@@ -32,57 +23,65 @@ type Props = {
...
+  const selectedUserChange = React.useCallback(
+    user => {
+      dispatch(LoginGen.createLoginError({error: null}))
+      setPassword('')
+      setSelectedUser(user)
+      if (props.loggedInMap.get(user)) {
+        props.onLogin(user, '')
+      }
+    },
+    [dispatch, setPassword, setSelectedUser, props]
```
Comment (chrisnojima) [inline shared/login/relogin/container.tsx]: same here

Comment (jzila) [reply shared/login/relogin/container.tsx]: lint yells at me if I don't include props, but I can destructure.

Comment (chrisnojima) [reply shared/login/relogin/container.tsx]: You can pass props.onlogin or whatever

File: shared/login/join-or-login/index.tsx (hunk 4 above)
Comment (jzila) [reply shared/login/join-or-login/index.tsx]: I think you're commenting on something outdated.

File: shared/login/relogin/container.tsx (hunk 8 above)
Comment (jzila) [reply shared/login/relogin/container.tsx]: > react-hooks/exhaustive-deps: React Hook React.useCallback has a missing dependency: 'props'. Either include it or remove the dependency array. However 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useCallback call and refer to those specific props inside React.useCallback.

**Comment by jzila** [comment]
final review?

**Pull request merged**